ddtrace/internal/telemetry/writer.py

```diff
@@ -162,6 +164,34 @@ def get_headers(self, request):
         return headers
 
 
+class TelemetryWriterModuleWatchdog(BaseModuleWatchdog):
+    def __init__(self) -> None:
+        self._new_imported: Set[str] = set()
+        self._initial = True
+        super().__init__()
+
+    def after_import(self, module: ModuleType) -> None:
+        module_path = origin(module)
+        self._new_imported.add(str(module_path))
+
+    def get_new_imports(self):
+        if self._initial:
+            try:
+                # On the first call, use sys.modules to cover all imports before we started. This is not
+                # done on __init__ because we want to do this slow operation on the writer's periodic call
+                # and not on instantiation.
+                new_imports = [str(origin(i)) for i in sys.modules.values()]
```

Comment:
My feeling is that we might need to handle at least `RuntimeError`, like done in https://github.com/P403n1x87/dd-trace-py/blob/acfee55cdec740ba4af79cedddda845580d2afa6/ddtrace/internal/module.py#L405, because `sys.modules` might mutate while iterated upon

Reply:
Done!